bindinfo/handle.go

```diff
@@ -738,12 +738,12 @@ func (h *BindHandle) newBindRecord(row chunk.Row) (string, *BindRecord, error) {
 func (h *BindHandle) setBindRecord(hash string, meta *BindRecord) {
 	newCache, err0 := h.bindInfo.Value.Load().(*bindCache).Copy()
 	if err0 != nil {
-		logutil.BgLogger().Warn("[sql-bind] BindHandle.setBindRecord", zap.Error(err0))
```

Comment:
we can use `With()` to create the logger, so no need to repeated add the zap Fields?

Reply:
I think that can be used on a case-by-case bases. Seems quite useful.

https://pkg.go.dev/go.uber.org/zap#Logger.With

Reply:
I assume this would be something like this?

Instead of
```go
logutil.BgLogger().Warn("[sql-bind] BindHandle.setBindRecord", zap.Error(err0))

// or

logutil.BgLogger().Warn("BindHandle.setBindRecord", zap.Error(err0), zap.String("category", "sql-bind"))
```

Use
```go
bindLog := logutil.BgLogger().With(zap.String("category", "sql-bind"))

...

bindLog.Warn("BindHandle.setBindRecord", zap.Error(err0))
```

Reply:
@dveeden do you want to finish this PR as-is, and leave my suggestions for future change? If so, I can give a lgtm

Reply:
@lance6716 Yes.